### PR TITLE
Fix listing nodes in scheduler

### PR DESCRIPTION
**Pull request by wojtek-t**
This is a followup on #6717 - that PR changed the logic which nodes are used by the scheduler (i.e. it started ignoring Unschedulability and NodeReady condition).
In this PR I'm trying to restore the previous behavior.

### CODE REVIEW

**Comment by wojtek-t** [comment]
I'm running e2e test - will post the results later.

**Comment by wojtek-t** [comment]
e2e tests are passing:
Ran 35 of 40 Specs in 1127.332 seconds
SUCCESS! -- 35 Passed | 0 Failed | 1 Pending | 4 Skipped I0415 14:57:36.580695   18430 driver.go:117] All tests pass

File: pkg/registry/minion/rest.go [hunk 1]
```diff
@@ -107,7 +107,8 @@ type ResourceGetter interface {
 // NodeToSelectableFields returns a label set that represents the object.
 func NodeToSelectableFields(node *api.Node) fields.Set {
 	return fields.Set{
-		"metadata.name": node.Name,
+		"metadata.name":      node.Name,
+		"spec.unschedulable": fmt.Sprint(node.Spec.Unschedulable),
```
Comment (lavalamp) [inline pkg/registry/minion/rest.go]: Ugh, did we really call this field Unschedulable? :( :( field names should always be in the positive IMO (should be Schedulable)

**Comment by lavalamp** [comment]
This LGTM, but can you add an integration test that verifies that nothing gets scheduled on an unschedulable node? This is functionality that we definitely need to keep working.

Thanks for cleaning up my mess!

**Comment by lavalamp** [comment]
I think I will merge this now, since this is important--and maybe you can do a follow-up PR with a test.

**Comment by wojtek-t** [comment]
@lavalamp thanks - will work on the test today

File: pkg/registry/minion/rest.go (hunk 1 above)
Comment (wojtek-t) [inline pkg/registry/minion/rest.go]: I completely agree, but I'm afraid it's too late now to change it :(

**Pull request merged**